src/code/entities/05_marin.asm

```diff
@@ -644,177 +656,215 @@ func_005_51BC::
-.variant3
+.variant3 ; looking at Marin
     db $48, OAM_GBC_PAL_0 | OAMF_PAL0
     db $4A, OAM_GBC_PAL_0 | OAMF_PAL0
-.variant4
+.variant4 ; looking at Marin, blinking
     db $48, OAM_GBC_PAL_0 | OAMF_PAL0
     db $4C, OAM_GBC_PAL_0 | OAMF_PAL0
 
 ; define sprites and there OAM Attributes in a list
-MarinDropGBCSpriteList::
+LinksBedMattressSpriteRect::
     ;  x    y    n°   OAM
     db $00, $00, $4E, OAM_GBC_PAL_6 | OAMF_PAL0
     db $00, $08, $4E, OAM_GBC_PAL_6 | OAMF_PAL0 | OAMF_XFLIP
 
-Data_005_5274::
+; List of variants for playing the animation of Link tossing and turning in bed.
+LinkInBedVariantsAnimation::
     db   $03, $03, $03, $03, $03, $04, $03, $04, $03, $03, $03, $02, $02, $02, $02, $02
     db   $00, $00, $01, $01, $00, $00, $01, $01, $00, $00, $01, $01, $00, $00, $01, $01
 
-func_005_5294::
+MarinLinkTossingHandler::
+    ; On countdown end, open the "What a relief!" dialog and move to the next state
     call GetEntitySlowTransitionCountdown         ;; 05:5294 $CD $FB $0B
-    jr   nz, .jr_52A4                             ;; 05:5297 $20 $0B
-
-    call_open_dialog Dialog001                    ;; 05:5299 $3E $01
+    jr   nz, .dialogEnd                           ;; 05:5297 $20 $0B
+    call_open_dialog Dialog001 ; "What a relief!" ;; 05:5299 $3E $01
     ld   [hl], $40                                ;; 05:529E $36 $40
     call IncrementEntityState                     ;; 05:52A0 $CD $12 $3B
     xor  a                                        ;; 05:52A3 $AF
+.dialogEnd
 
-.jr_52A4
     rra                                           ;; 05:52A4 $1F
     rra                                           ;; 05:52A5 $1F
     and  $1F                                      ;; 05:52A6 $E6 $1F
     ld   e, a                                     ;; 05:52A8 $5F
     ld   d, b                                     ;; 05:52A9 $50
-    ld   hl, Data_005_5274                        ;; 05:52AA $21 $74 $52
+    ld   hl, LinkInBedVariantsAnimation           ;; 05:52AA $21 $74 $52
     add  hl, de                                   ;; 05:52AD $19
     ld   a, [hl]                                  ;; 05:52AE $7E
+    ; fallthrough
 
-func_005_52AF::
+; Render Link in bed
+;
+; Inputs:
+;   a    sprite variant
+RenderLinkInBedSprites::
     ldh  [hActiveEntitySpriteVariant], a          ;; 05:52AF $E0 $F1
     ld   a, $38                                   ;; 05:52B1 $3E $38
     ldh  [hActiveEntityPosX], a                   ;; 05:52B3 $E0 $EE
     ldh  [hLinkPositionX], a                      ;; 05:52B5 $E0 $98
     ld   a, $34                                   ;; 05:52B7 $3E $34
     ldh  [hActiveEntityVisualPosY], a             ;; 05:52B9 $E0 $EC
     ldh  [hLinkPositionY], a                      ;; 05:52BB $E0 $99
+
     ld   a, $02                                   ;; 05:52BD $3E $02
     ldh  [hLinkInteractiveMotionBlocked], a       ;; 05:52BF $E0 $A1
+
     ld   a, LINK_ANIMATION_STATE_NO_UPDATE        ;; 05:52C1 $3E $FF
```

Comment:
While this says "no update", I think this actually hides link completely. Might need a test if this only hides link if you directly set this on room load or if you can always use this to hide the link sprite.

Reply:
Yes, it does hide Link completely. It is also used when Link is falling down a pit, at the end of the animation, when he is no longer visible.

I renamed it to LINK_ANIMATION_STATE_HIDDEN.